.github/workflows/ci.yaml

```diff
@@ -145,22 +145,27 @@ jobs:
 
       # Publish a syncer image to the local registry
       - uses: imjasonh/setup-ko@v0.4
-      - run: KO_DOCKER_REPO=localhost:5000 ko publish ./cmd/syncer --insecure-registry -t $(git rev-parse --short "$GITHUB_SHA") > /tmp/syncer-image
+      - run: KO_DOCKER_REPO=${KIND_REGISTRY} ko publish ./cmd/syncer --insecure-registry -t $(git rev-parse --short "$GITHUB_SHA") > /tmp/syncer-image
 
       - run: |-
           export LOG_DIR=/tmp/e2e/shared-server/artifacts &&
           mkdir -p ${LOG_DIR} &&
           NO_GORUN=1 ./bin/test-server \
           > ${LOG_DIR}/kcp.log 2>&1 &
+          echo $! > /tmp/kcp.pid
+
       - run: |-
+          kind get kubeconfig > /tmp/kind-kubeconfig
           ARTIFACT_DIR=/tmp/e2e \
           PATH="${PATH}:$(pwd)/bin/" \
-          TEST_ARGS="-args --use-default-kcp-server --syncer-image=$(cat /tmp/syncer-image) --pcluster-kubeconfig=${KUBECONFIG}" \
+          TEST_ARGS="-args --use-default-kcp-server --syncer-image=$(cat /tmp/syncer-image) --pcluster-kubeconfig=/tmp/kind-kubeconfig" \
```

Comment:
$KUBECONFIG is not set by anything. I added retrieving the kubeconfig above and storing it to /tmp/kind-kubeconfig.